packages/nimble-components/src/rich-text/viewer/tests/rich-text-viewer.spec.ts

```diff
@@ -81,4 +89,209 @@ describe('RichTextViewer', () => {
 
         await disconnect();
     });
+
+    describe('User mention', () => {
+        beforeEach(async () => {
+            await connect();
+        });
+
+        afterEach(async () => {
+            await disconnect();
+        });
+
+        it('adding mention configuration converts the absolute like matching the scheme with pattern to mention node', async () => {
+            element.markdown = '<user:1>';
+
+            expect(pageObject.getRenderedMarkdownTagNames()).toEqual([
+                'P',
+                `${anchorTag}`.toUpperCase()
+            ]);
+            expect(pageObject.getRenderedMarkdownLastChildContents()).toBe(
+                'user:1'
+            );
+            await waitForUpdatesAsync();
+
+            const userMention = document.createElement(
+                richTextMentionUsersTag
+            ) as RichTextMentionUsers;
+            userMention.pattern = '^user:(.*)';
+
+            element.appendChild(userMention);
+            await waitForUpdatesAsync();
+
+            expect(pageObject.getRenderedMarkdownTagNames()).toEqual([
+                'P',
+                `${richTextMentionUsersViewTag}`.toUpperCase()
+            ]);
+            expect(
+                pageObject.getRenderedMarkdownLastChildAttribute(
+                    'mention-label'
+                )
+            ).toEqual('1');
+        });
+
+        it('adding mention mapping renders the mapped display name', async () => {
+            element.markdown = '<user:1>';
+
+            expect(pageObject.getRenderedMarkdownTagNames()).toEqual([
+                'P',
+                `${anchorTag}`.toUpperCase()
+            ]);
+            expect(pageObject.getRenderedMarkdownLastChildContents()).toBe(
+                'user:1'
+            );
+            await waitForUpdatesAsync();
+
+            const userMention = document.createElement(
+                richTextMentionUsersTag
+            ) as RichTextMentionUsers;
+            userMention.pattern = '^user:(.*)';
+
+            const mappingUser = document.createElement(
+                mappingUserTag
+            ) as MappingUser;
+            mappingUser.key = 'user:1';
+            mappingUser.displayName = 'username1';
+            userMention.appendChild(mappingUser);
+
+            element.appendChild(userMention);
+            await waitForUpdatesAsync();
```

Comment:
To reduce duplication, I would pull this out into a helper function that you call from each of these tests:
```TS
async function appendUserMentionConfiguration(element: RichTextViewer, userKey: string | undefined, displayName: string | undefined) {
   const userMention = document.createElement(
        richTextMentionUsersTag
    ) as RichTextMentionUsers;
    userMention.pattern = '^user:(.*)';

    if (userKey || displayName) {        
        const mappingUser = document.createElement(
            mappingUserTag
        ) as MappingUser;
        mappingUser.key = userKey ?? '';
        mappingUser.displayName = displayName ?? '';
        userMention.appendChild(mappingUser);
    }
    element.appendChild(userMention);
    await waitForUpdatesAsync();
}

Reply:
I moved this to a helper method and used whenever needed as suggested.